ghcide/src/Development/IDE/Spans/AtPoint.hs

```diff
@@ -308,6 +311,24 @@ atPoint IdeOptions{} (HAR _ hf _ _ (kind :: HieKind hietype)) (DKMap dm km) env
             UnhelpfulLoc {} | isInternalName name || isSystemName name -> Nothing
             _ -> Just $ "*Defined " <> printOutputable (pprNameDefnLoc name) <> "*"
 
+        prettyInstances :: (Either ModuleName Name, IdentifierDetails hietype) -> IO (Maybe T.Text)
+        prettyInstances (Right n, _) =
+            fmap (wrapHaskell . T.unlines . fmap printOutputable) <$> instancesForName
+            where
+                instancesForName :: IO (Maybe [ClsInst])
+                instancesForName = runMaybeT $ do
+                    typ <- MaybeT . pure $ lookupNameEnv km n >>= tyThingToType
```

Comment:
It looks like we look things up repeatedly in `km`. Should we do this once and then just pass in the `TyThing` to all the places that need it?

Reply:
I am not sure what you mean. I think the lookup only happens once per `Name` and we need to look it up for each `Name`

Reply:
I mean in the other functions we call around here. Probably not a huge deal